data/account/participationRegistry.go

```diff
@@ -767,6 +770,9 @@ func (db *participationDB) GetStateProofSecretsForRound(id ParticipationID, roun
 	if err != nil {
 		return StateProofSecretsForRound{}, err
 	}
+	if partRecord.StateProof == nil {
+		return StateProofSecretsForRound{}, ErrStateProofVerifierIsNil
```

Comment:
Provide the ID for debugging:
```suggestion
		return StateProofSecretsForRound{}, fmt.Errorf("%s: for participation ID %s", ErrStateProofVerifierNotFound, id)
```